### PR TITLE
Error handling #262

### LINKED ISSUE #262: Error handling for IMS
- [x] Remove the console.logs
- [x] Create handle error state 
- [ ] Test in scigateway
- [x] Global error handling for the use Query (get requests)
- [x]  Add the handleIMS_APIError to the catch of the mutateAsyncs as most of the error handling is done in the dialogs
- [x] refersh token

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

- Add global error handler for the get requests
- Add global retry handler 
- Remove console.log for the react query functions 
- Add refresh token logic

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] Check requests behave sensibly including giving error messages when they fail (raise them in the API to trigger)

## Agile board tracking

Closes #262

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
waiting on auth in the backend

File: src/handleIMS_APIError.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { AxiosError } from 'axios';
+import log from 'loglevel';
+import {
+  NotificationType,
+  InvalidateTokenType,
+} from './state/actions/actions.types';
+import { ErrorParsing, MicroFrontendId } from './app.types';
+
+const handleIMS_APIError = (error: AxiosError, broadcast = true): void => {
+  const message = error.response?.data
```
Comment (joshuadkitenge) [inline src/handleIMS_APIError.ts:10]: Fix this

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Two minor things, and some conflicts to resolve. Tested with https://github.com/ral-facilities/inventory-management-system-api/pull/169, error messages worked well and redirect to login when the token expires works as well.

File: src/catalogue/items/catalogueItemsDialog.component.test.tsx [hunk 2]
```diff
@@ -672,11 +673,7 @@ describe('Catalogue Items Dialog', () => {
       const saveButton = screen.getByRole('button', { name: 'Save' });
       await user.click(saveButton);
 
-      await waitFor(() => {
-        expect(
-          screen.getByText('Please refresh and try again')
-        ).toBeInTheDocument();
-      });
+      expect(handleIMS_APIError).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/catalogue/items/catalogueItemsDialog.component.test.tsx:676]: The handleIMS_APIError components has the tests. The handleIMS_APIError is only for the uncaught errors (500)

**Comment by joshuadkitenge** [comment]
Add error handling for the moveto and copyto

File: src/catalogue/items/catalogueItemsDialog.component.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (joelvdavies) [reply src/catalogue/items/catalogueItemsDialog.component.test.tsx:676]: Yes but it doesn't check what it is sending handleIMS_APIError. It was more for ensuring the appropriate message and broadcast is sent and not something else, But then again none of them specify broadcast at the moment and the matcher for the error would probably have to be expect.anything() anyway, so its probably alright for now.

**Comment by joshuadkitenge** [comment]
do the todos in the api functions

**Comment by joshuadkitenge** [comment]
I copy loads of catalogue categories and then I was trying to delete them it done this when it triggered a refresh @joelvdavies 


error handle bug

**Review by VKTB — CHANGES_REQUESTED** [review]
The requests to the `/units` API endpoint fail with `403 Not authenticated`. I think that the JWT access token is not supplied.

**Review by VKTB — APPROVED** [review]

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**